src/core_functions/scalar/list/array_slice.cpp

```diff
@@ -163,6 +163,18 @@ static void ExecuteConstantSlice(Vector &result, Vector &str_vector, Vector &beg
                                  optional_ptr<Vector> step_vector, const idx_t count, SelectionVector &sel,
                                  idx_t &sel_idx, optional_ptr<Vector> result_child_vector, bool begin_is_empty,
                                  bool end_is_empty) {
+
+	// check all this nullness early
```

Comment:
This early-out looks good to me! `sel_valid` can only ever be `true` if none of this nullness applies. All other cases are also never reached if any of these is not valid.